### ISSUE TITLE
scrobbling support for more providers than just last.fm

### ISSUE THREAD

**Issue by Clementine-Issue-Importer** [clementine-player/Clementine#1]
_From [mattl-lists@fsf.org](https://code.google.com/u/100106606012998798867/) on February 22, 2010 22:57:45_

Can you offer people the chance to enter any host name they wish for
scrobbling? So, if people wish to use turtle.libre.fm instead of
post.audioscrobbler.com they can?

_Original issue: http://code.google.com/p/clementine-player/issues/detail?id=1_

**Comment by grovve** [comment]
In the meantime, the solution that is available to scrobble in both in libre.fm and last.fm using clementine and windows is:
1) Modify the "hosts" file in windows and redirect the audioscrobbler to http://turtle.libre.fm
2) Activate connection with last.fm in the libre.fm website (menu Edit > "Connections to other services"). Add the desire lastfm profile where you want to scrobble.
3) Add your libre.fm credentials in the lastfm tab under Preferences > Internet Providers in Clementine settings.

I use that set up and I am currently scrobbling in both websites. Hope it helps until we get a better solution!

**Comment by mlncn** [comment]
Adding my voice to the requests and offers to support development of an easy way to scrobble to libre.fm instead of or in addition to last.fm.

**Comment by mattl** [comment]
I just upgraded my laptop, and I'd be using Clementine right now if Libre.fm support was available.

If there's anything *we* can do to make this happen, please let me know.

Libre.fm founder ;)

**Comment by Chocobozzz** [comment]
Clementine uses liblastfm. I can embed custom liblastfm to the Clementine's 3rdparty directory and force user to compile it with cmake. The custom liblastfm would add a parameter to specify the url where scobbling (something like that).

@hatstand @davidsansome @ArnaudBienner do you have any objections ?

If no I'll try to work on it.

**Comment by obsidiannonparity** [comment]
That sounds like it should work @Chocobozzz 

It would be great to have this support in Clementine

**Comment by Chocobozzz** [comment]
Yes it works but add a static library in which we modified three lines while it exists in practically all Linux distributions is not really acceptable. 

So I [made a PR in the liblastfm github repo](https://github.com/lastfm/liblastfm/pull/30) but they don't seem to want to merge it.

Wait & see.

**Comment by ar-jan** [comment]
Hoping for this feature to go in. I'd rather stick with Clementine than start using another client with built-in support for libre.fm.

**Comment by nohponex** [comment]
> Hoping for this feature to go in. I'd rather stick with Clementine than start using another client with built-in support for libre.fm.

Me too

**Comment by jordanbsanders** [comment]
> Hoping for this feature to go in. I'd rather stick with Clementine than start using another client with built-in support for libre.fm.

+1

**Comment by JulianVolodia** [comment]
I think it should be closed ASAP or devided info smaller issues, Guys! :)

**Comment by ar-jan** [comment]
Looks like the workaround of redirecting last.fm scrobbling to libre.fm via the hosts file stopped working from version 1.3.0 onwards :(

If you try the Last.fm `Sign in` button under Preferences, with `213.138.110.197 post.audioscrobbler.com` in the hosts file, you only get a popup saying "Last.fm authentication faile[d]".

**Comment by mattl** [comment]
@ar-jan Libre.fm was down some a few days, so possibly related to that?

**Comment by ar-jan** [comment]
@mattl I think it's unrelated: my scrobble history in libre.fm ends about 2 months ago, and I tried logging in again yesterday, but failed. I recall seeing something about a new login method for last.fm, so my guess is that when this was implemented for last.fm, the hosts file redirect option stopped working.

**Comment by JulianVolodia** [comment]
@ar-jan and @mattl isn't it's another (new) issue? It not connected with "scrobbling support for more providers than just last.fm", have higher priority and it's bug (not prior-low, enhancement).

Added here #5433
Reopening old issue will scary new people to participe in project.

What about main issue? Is there anything to make?

**Comment by ar-jan** [comment]
Yeah I guess, I wasn't sure if the hosts redirect workaround is officially supported so I made it a comment. I added some info to the new issue.

**Comment by stephenjudge** [comment]
I'd like to suggest when finally adding Libre.fm compatibility, that you also as ListenBrainz compatibility. ListenBrainz is a new audio scrobbling type project from the MusicBrainz community. You can add ListenBrainz support with their [Last.FM Compatible API](https://listenbrainz.readthedocs.io/en/latest/dev/api-compat.html)

**Comment by johan-boule** [comment]
In the meantime, you can edit your /etc/hosts file as described here: https://listenbrainz.org/lastfm-proxy

**Comment by flip-bs** [comment]
it seems that this this fork of Clementine support listenbrianz 
Strawberry Music Player https://strawbs.org/
https://github.com/jonaski/strawberry

**Comment by JulianVolodia** [comment]
Hello
Could you clean up here @jee-r , @johan-boule , @stephenjudge ?

Before doing below check if issue is quite recent, and maybe it was already fixed.

Great to have:
- summary *what to do* here when someone get this item to work with,
- fulfill issue again, even changing subject to have it clear with corespondence with current state of issue, leaving it in *great condition* in backlog for future work when someone have time.

Will appreciate if some important (like workarounds, recomendation for fixes, pre-investigation info) will be left with mark with ~shattering~ with `~this~` for fast reference, but best is to leave issue *most compressed as you could*.

Thanks in advance.

**Comment by johan-boule** [comment]
> Hello Could you clean up here [@jee-r](https://github.com/jee-r) , [@johan-boule](https://github.com/johan-boule) , [@stephenjudge](https://github.com/stephenjudge) ?
> 
> Before doing below check if issue is quite recent, and maybe it was already fixed.
> 
> Great to have:
> 
>     * summary _what to do_ here when someone get this item to work with,
> 
>     * fulfill issue again, even changing subject to have it clear with corespondence with current state of issue, leaving it in _great condition_ in backlog for future work when someone have time.
> 
> 
> Will appreciate if some important (like workarounds, recomendation for fixes, pre-investigation info) will be left with mark with ~shattering~ with `~this~` for fast reference, but best is to leave issue _most compressed as you could_.
> 
> Thanks in advance.

Sorry, I cannot do any cleanup because I have only learned standard HTML, not GitHub/Microsoft mark down syntax.
